src/main/java/org/apache/commons/codec/digest/DigestUtils.java

```diff
@@ -1207,6 +1236,43 @@ public static MessageDigest updateDigest(final MessageDigest digest, final File
         }
     }
 
+    /**
+     * Reads through a RandomAccessFile and updates the digest for the data
+     * using non-blocking-io (NIO)
+     *
+     * @param digest The MessageDigest to use (e.g. MD5)
+     * @param data   Data to digest
+     * @return the digest
+     * @throws IOException On error reading from the stream
+     * @since 1.14
+     */
+    public static MessageDigest updateDigestNio(final MessageDigest digest, final RandomAccessFile data) throws IOException {
```

Comment:
Don't use implementation details in method names, so -1 to the "Nio" postfix.

Reply:
I've added the prefix here for consistency with the other overload that takes a `File`.